src/components/StyledReactSelect/StyledReactSelect.tsx

```diff
@@ -0,0 +1,52 @@
+import { ComponentProps, useMemo } from 'react'
+import Select, { components, OptionProps } from 'react-select'
+import { ChevronDownIcon, ThickCheckIcon } from '@radix-ui/themes'
+import { getStylesConfig, getThemeConfig } from './StyledReactSelect.styles'
+
+export function StyledReactSelect<Option>(
+  props: ComponentProps<typeof Select<Option>>
+) {
+  const styles = useMemo(() => getStylesConfig<Option>(), [])
```

Comment:
Can this be moved to just before the component function? `useMemo` won't be needed then.

Reply:
The only reason is passing that generic type, was getting TS errors without it :(